R/pkg/R/DataFrame.R

```diff
@@ -3905,6 +3905,18 @@ setMethod("rollup",
             groupedData(sgd)
           })
 
+isTypeAllowed <- function(x) {
+  if (is.character(x)) {
+    TRUE
+  } else if (is.list(x)) {
```

Comment:
how is `list` going to be used? for instance, do we need to recurse into to check the type of items in the list?

Reply:
@felixcheung Thank you very much for your review! I don't think database will use nested list for sql hints, but I can change the code to 
```
isTypeAllowedForSqlHint <- function(x) {
  if (is.character(x)) {
    TRUE
  } else if (is.list(x)) {
    all(sapply(x, isTypeAllowedForSqlHint))
  } else if (is.numeric(x)) {
    TRUE
  } else {
    FALSE
  }
}
```
Does this look good to you?

Reply:
in R list could have a list too, not sure if we should recurse indefinitely
```
> a <- list(list(a = 1, b = 2))
> a
[[1]]
[[1]]$a
[1] 1

[[1]]$b
[1] 2

```

Reply:
I guess we should make the nested list work, but in real usage, only one level of list will be used in sql hint and the code will stop after checking the first level of list.

Reply:
I'd actually prefer to be more restrictive.. maybe just one level, unless we know of a real need for nested list?

Reply:
Thanks. I will make changes.